kyaml/kio/pkgio_reader.go

```diff
@@ -162,6 +165,11 @@ type LocalPackageReader struct {
 
 	// SetAnnotations are annotations to set on the Resources as they are read.
 	SetAnnotations map[string]string `yaml:"setAnnotations,omitempty"`
+
+	// ignoreFilesMatcher is used by the Read function to keep track
+	// of the .krmignore files encountered while walking the files in the
+	// package.
+	ignoreFilesMatcher *ignoreFilesMatcher
```

Comment:
seems like this could just be declarared/allocated inline?

Reply:
I have moved it into the `Read` function and pass it into the other functions that need it.